apps/testingaccessibility/src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,111 @@
+import NextAuth from 'next-auth'
+import GithubProvider from 'next-auth/providers/github'
+import GoogleProvider from 'next-auth/providers/google'
+import jwt from 'jsonwebtoken'
+import {JWT} from 'next-auth/jwt'
+
+export default NextAuth({
+  secret: process.env.NEXTAUTH_SECRET,
+  jwt: {
+    secret: process.env.NEXTAUTH_SECRET,
+    encode: async ({secret, token, maxAge}) => {
+      const encodedToken = jwt.sign(
+        {
+          ...token,
+          iat: Date.now() / 1000,
+          exp: Math.floor(Date.now() / 1000) + 24 * 60 * 60,
+          'https://hasura.io/jwt/claims': {
+            'x-hasura-allowed-roles': ['user'],
+            'x-hasura-default-role': 'user',
+            'x-hasura-role': 'user',
+            'x-hasura-user-id': token?.id,
+          },
+        } || '',
+        secret,
+        {algorithm: 'HS512'},
+      )
+
+      return encodedToken
+    },
+    decode: async (params) => {
+      const verify = jwt.verify(params.token || '', params.secret)
+      return verify as JWT
+    },
+  },
+  providers: [
+    GithubProvider({
+      clientId: process.env.GITHUB_ID,
+      clientSecret: process.env.GITHUB_SECRET,
+    }),
+    GoogleProvider({
+      clientId: process.env.GOOGLE_ID,
+      clientSecret: process.env.GOOGLE_SECRET,
+      authorization: {
+        params: {
+          prompt: 'consent',
+          access_type: 'offline',
+          response_type: 'code',
+        },
+      },
+    }),
+  ],
```

Comment:
google and github providers (keys in vercel env) seems like it would have significant coverage. We can add others. LinkedIn would be another strong contender.

One thing to note is that each provider is a unique account, so they will be identified by their provider account. They don't blend.